var/spack/repos/builtin/packages/tiramisu/package.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import llnl.util.filesystem as fs
+from spack.package import *
+
+
+class Tiramisu(CMakePackage, PythonExtension):
+    """Tiramisu is a polyhedral compiler for dense and sparse deep learning
+    and data parallel algorithms.It provides a simple C++ API for expressing
+    algorithms and how these algorithms should be optimized by the compiler."""
+
+    homepage = "http://tiramisu-compiler.org"
+    url = "https://github.com/Tiramisu-Compiler/tiramisu"
+    git = "https://github.com/Tiramisu-Compiler/tiramisu.git"
+
+    maintainers = ["wraith1995"]
+
+    generator = "Ninja"
+
+    version("master", branch="master")
+    version("2023-2-3", commit="73a9cec72e08d4dfe5e8c66da33139008124a4fa")
+
+    variant("python", default=True, description="Install python bindings.")
+    variant("cuda", default=False, description="Enable Cuda Code Generation.")
+    extends("python", when="+python")
+    variant(
+        "debug",
+        default="0",
+        description="Set the debug level.",
+        multi=False,
+        values=("0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10"),
+    )
+
+    depends_on("cmake@3.24:", type="build")
+    depends_on("ninja", type="build")
+
+    depends_on("halide@14.0.0", type=("build", "link", "run"))
+    depends_on("isl", type=("build", "link", "run"))
+    depends_on("cuda", when="+cuda", type=("build", "link", "run"))
+
+    depends_on("python@3.8:", type=("build", "link", "run"), when="+python")
+    depends_on("py-pybind11@2.6.2:", type=("build", "link", "run"), when="+python")
+    depends_on("py-numpy", type=("build", "run"), when="+python")
+    depends_on("py-cython", type=("run"), when="+python")
+
+    def cmake_args(self):
+        spec = self.spec
+        llvmdir = spec["llvm"].prefix.bin
+        args = [
+            self.define("HALIDE_LIB_DIRECTORY", spec["halide"].prefix.lib),
+            self.define("LLVM_CONFIG_BIN", llvmdir),
+            self.define("HALIDE_SOURCE_DIRECTORY", spec["halide"].prefix),
+            self.define("ISL_INCLUDE_DIRECTORY", spec["isl"].prefix.include),
+            self.define("ISL_LIB_DIRECTORY", spec["isl"].prefix.lib),
+            self.define_from_variant("DEBUG_LEVEL", "debug"),
+            self.define_from_variant("WITH_PYTHON_BINDINGS", "python"),
+            self.define_from_variant("USE_GPU", "cuda"),
+            self.define("PYBIND11_USE_FETCHCONTENT", False),
+            self.define("USE_FLEXNLP", False),
+        ]
+        if "+python" in spec:
+            args += [
+                self.define("Tiramisu_INSTALL_PYTHONDIR", python_platlib),
+                self.define("Python3_EXECUTABLE", spec["python"].command.path),
+            ]
+        return args
+
+    def build(self, pkg, spec):
```

Comment:
Is there a reason to use cmake instead of make for the build/install phases? I haven't seen this in any other package.

Reply:
I did this because I knew it worked and the defaults did not. I am not sure why. I can investigate further if required, but I do agree it is really weird.

Reply:
If you can get me an error message that may help

Reply:
Sure. I'd appreciate that though I don't recall them being particularly illuminating. IIRC it tries to build more than it should, but not the dependencies for these things. Let me go erase those lines and upload some logs.

Reply:
I realized the problem. This is quite silly. I'll push a solution in a few mins.

Reply:
Okay. Basically, I was not properly specifying the targets, however, now I've found a bug in the cmake files. Thanks for prompting me to look at this agian. I will fix that, get it merged in, and then ping you when done.

Reply:
@adamjstewart It is done. 

The short short version is that a standard use of the cmake package now works.

The short version is that CMake implicitly made the install targets depend on all targets and some targets (benchmarks mainly) didn't work for our of source builds due to some hacky cmake. I've fixed the first issue and decided not to expose benchmarks here until I have time to fix them too.